App/Model/Fast.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Model;
+
+use App\Enums\FastType;
+use App\Enums\Status;
+use App\Interface\FastModelInterface;
+use DateTime;
+use Exception;
+use JetBrains\PhpStorm\ArrayShape;
+use JsonSerializable;
+
+
+class Fast implements FastModelInterface, JsonSerializable
+{
+    protected array $dates = [
+        'start',
+        'end'
+    ];
+
+    protected array $fillable = [
+        'start',
+        'status',
+        'end',
+        'type',
+        'elapsedTime'
+    ];
+
+    public function __construct(
+        protected string $start = '',
+        protected int    $status = 1,
+        protected string $end = '',
+        protected int    $type = 0,
+        protected string $elapsedTime = '')
+    {
+    }
+
+    /**
+     * @param $parameter
+     * @return DateTime|int|string
+     * @throws Exception
+     */
+    public function __get($parameter): DateTime|int|string
+    {
+        if (in_array($parameter, $this->dates)) {
+            return new DateTime($this->{$parameter});
+        }
+
+        return $this->$parameter;
+    }
+
+    /**
+     * Global params setter for fillable properties of Fast class.
+     * @param array $params
+     * @throws Exception
+     */
+    public function set(array $params)
+    {
+        foreach ($params as $param => $value) {
+            if (!in_array($param, $this->fillable)) throw new Exception("Cannot set property {$param} of " . get_class($this));
+            $this->$param = $value;
+        }
+    }
+
+    /**
+     * @return string
+     * @throws \ReflectionException
+     */
+    public function print(): string
+    {
+        return "
+--------------------------------------
```

Comment:
This one also should be formatted :)

Reply:
I don't understand can you please specify. This is written this way because this is the exact format i want to be printed into the console.